Sources/EpoxyCore/SwiftUI/LayoutUtilities/MeasuringUIViewRepresentable.swift

```diff
@@ -0,0 +1,82 @@
+// Created by eric_horacek on 6/22/22.
+// Copyright © 2022 Airbnb Inc. All rights reserved.
+
+import SwiftUI
+
+// MARK: - MeasuringUIViewRepresentable
+
+/// A `UIViewRepresentable` that uses a `SwiftUIMeasurementContainer` wrapping its represented
+/// `UIView` to report its size that fits a proposed size to SwiftUI.
+///
+/// Supports iOS 13-15 using the private `_overrideSizeThatFits(…)` method and iOS 16+ using the
+/// `sizeThatFits(…)` method.
+///
+/// - SeeAlso: ``SwiftUIMeasurementContainer``
+public protocol MeasuringUIViewRepresentable: UIViewRepresentable
+  where
+  UIViewType == SwiftUIMeasurementContainer<Self, View>
+{
+  /// The `UIView` that's being measured by the enclosing `SwiftUIMeasurementContainer`.
+  associatedtype View: UIView
+
+  /// The sizing strategy of the represented view.
+  ///
+  /// To configure the sizing behavior of the `View` instance, call `sizing` on this `View`, e.g.:
+  /// ```
+  /// myView.sizing(.intrinsicSize)
+  /// ```
+  var sizing: SwiftUIMeasurementContainerStrategy { get set }
+}
+
+// MARK: Extensions
+
+extension MeasuringUIViewRepresentable {
+  /// Returns a copy of this view with its sizing strategy updated to the given `sizing` value.
+  public func sizing(_ strategy: SwiftUIMeasurementContainerStrategy) -> Self {
+    var copy = self
+    copy.sizing = strategy
+    return copy
+  }
+}
+
+// MARK: Defaults
+
+extension MeasuringUIViewRepresentable {
+  public func _overrideSizeThatFits(
```

Comment:
is this an override function, or it is just expected that SwiftUI / UIKit will call this method?

Reply:
SwiftUI will just call this for us 👍 — it just has a weird name